### DISCUSSION TITLE
Unable to run OpenFGA Playground on k8s pod using helm chart

### DISCUSSION THREAD

**Discussion by shubhamsaininnl** [openfga/community#200] [Q&A] [answered]
Hi @team, 

I am using openfga helm chart and deploying on my own k8s cluster but i am getting error **play.fga.dev refused to connect.**

**Comment by shubhamsaininnl** [comment]
![openfga](https://github.com/openfga/openfga/assets/23218125/d4eac69a-dee1-46f7-b739-a699ff5e7a77)

**Comment by shubhamsaininnl** [comment]
hi @team, anyone can check this pls

**Comment by rhamzeh** [comment, marked as answer]
Hi @shubhamsaininnl - Playground will refuse to load at any url that is not localhost:3000 at the moment. We enforce that in the CSP. We have plans for a new version of the Playground that can be self hosted but that is still low priority for us at the moment.

What I would recommend is using our [VS Code extension](https://github.com/openfga/vscode-ext#installation) + the [CLI](https://github.com/openfga/cli/#installation) to manage your model.

The VS Code extension currently lacks the validations of the Playground, but we're working on that.

**Comment by HarshVoidUpparwal** [comment]
@rhamzeh can you share a plan or timeline on releasing the self-hosted playground?

**Comment by aaguiarz** [comment]
Hi @HarshVoidUpparwal 

We don't have a timeline yet. It implies rewriting the playground code and making it open source, and it's not a small effort.

Did you try the VS Code + CLI workflow? The VS Code implementation now has the same validations as the Playground, and the [CLI workflow to test your model](https://github.com/openfga/cli/#run-tests-on-an-authorization-model) is pretty good. 

What do you find lacking?
